cmd/security-bootstrap-redis/res/configuration.toml

```diff
@@ -14,24 +14,35 @@
 #################################################################################
 
 [Writable]
-  LogLevel = "ERROR"
+LogLevel = 'ERROR'
 
 [Service]
-  BootTimeout = 30000
+BootTimeout = 30000
+
+[Logging]
+EnableRemote = false
+File = ''
 
 [SecretStore]
-  Host = "edgex-vault"
-  Port = 8200
-  Path = "/v1/secret/edgex/redis"
-  RootCaCertPath = "/tmp/edgex/secrets/ca/ca.pem"
-  TokenFile = "/tmp/edgex/secrets/edgex-redis/secrets-token.json"
-  PasswordFile = "/tmp/edgex/secrets/edgex-redis/redis5-password"
-  ServerName = "edgex-vault"
-  AdditionalRetryAttempts = 10
-  RetryWaitPeriod = "1s"
+Host = 'localhost'          ## Override in environment variables, if necessary
+ServerName = 'edgex-vault'  ## Needed for TLS CN
```

Comment:
Actually, in the snapcraft.yaml,

```
      # install the json config files for security-secrets-setup (for kong
      # and vault PKI) and security-file-token-provider
      #
      # note that the file contents are processed using the real
      # value of $SNAP_DATA using jq in the install hook
      cp "./cmd/security-secrets-setup/res/pkisetup-vault.json" \
        "$SNAPCRAFT_PART_INSTALL/config/security-secrets-setup/res/pkisetup-vault.json"
      cp "./cmd/security-secrets-setup/res/pkisetup-kong.json" \
        "$SNAPCRAFT_PART_INSTALL/config/security-secrets-setup/res/pkisetup-kong.json"
      cp "./cmd/security-file-token-provider/res/token-config.json" \
        "$SNAPCRAFT_PART_INSTALL/config/security-file-token-provider/res/token-config.json"
```

which refers to pkisetup-vault
```
    "x509_tls_server_parameters": {
        "tls_host": "edgex-vault",
        "tls_domain": "local",
        "tls_c": "US",
        "tls_st": "CA",
        "tls_l": "San Francisco",
        "tls_o": "Vault"
    }
```
Causes a cert that looks like this:
```
        Subject: C = US, ST = CA, L = San Francisco, O = edgex-vault, OU = Vault, CN = edgex-vault
            X509v3 Subject Alternative Name: 
                DNS:edgex-vault, DNS:edgex-vault.local, email:admin@local
```

Thus, it is reasonable to expect a certificate coming back with "edgex-vault" even in the localhost case, because the snap doesn't override it and change the certificate commonname and SAN to "localhost".

Reply:
> For non-docker, which are the defaults, this is also `localhost`

@lenny-intel please elaborate as I don't understand what you're saying

Reply:
Also @bnevis-i I don't follow

Reply:
@andresrinivasan, I am trying to prove to @lenny-intel that the Vault will present an "edgex-vault" identified TLS certificate, even on localhost, and that ServerName = 'edgex-vault' is correct.  (But it doesn't really matter if we override it every place it is used.)

Reply:
I was unaware that the Cert used `edgex-vault` for the host name and that also worked with the non-docker deployment. Nice.